lib/synapse/haproxy.rb

```diff
@@ -811,5 +840,40 @@ def construct_name(backend)
 
       return name
     end
+
+    def update_state_file(watchers)
+      log.info "Writing state file"
+
+      timestamp = Time.now.to_i
+
+      # Remove stale backends
+      @seen.each do |watcher_name, backends|
+        backends.each do |backend_name, backend|
+          ts = backend.fetch('timestamp', 0)
+          delta = (timestamp - ts).abs
```

Comment:
Are you sure this should be abs? We don't want to expire things that are already in the file when we start up (e.g. the state file says timestamp 200 and we start at time 0)